concrete/src/User/UserInfo.php

```diff
@@ -330,104 +381,106 @@ public function getUserObject()
     /**
      * @param array $data
      *
-     * @return bool|null
-     */
-    public function update($data)
-    {
-        $db = $this->connection;
-        if ($this->getUserID()) {
-            $ux = $this->getUserObject();
-            $uName = $this->getUserName();
-            $uEmail = $this->getUserEmail();
-            $uHasAvatar = $this->hasAvatar();
-            $uTimezone = $this->getUserTimezone();
-            $uDefaultLanguage = $ux->getUserDefaultLanguage();
+     * @return bool|null returns false if the record has not been saved, null if the password confirmation failed, true otherwise
+     */
+    public function update(array $data)
```

Comment:
Let's update only the fields specified in `$data`, avoid code duplication, get rid of deprecated methods